### PR TITLE
Adding documentation and information

### PR DESCRIPTION
**Pull request by EmpireWorld**
Adding an installation and Usage example from the babel handbook https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#babel-traverse

Doc PR: yes

Adding some usage guides and installation.

Adding a link to the babel handbook.

Cheers. :clap: :tada:

### CODE REVIEW

File: packages/babel-traverse/README.md [hunk 1]
```diff
@@ -1 +1,30 @@
...
+const code = `function square(n) {
+  return n * n;
+}`;
+
+const ast = babylon.parse(code);
+
+traverse(ast, {
+  enter(path) {
+    if (
+      path.node.type === "Identifier" &&
```
Comment (loganfsmyth) [inline packages/babel-traverse/README.md]: FYI you can also do `if (path.isIdentifier({name: "n"})) {` to shorten up this example.

Comment (EmpireWorld) [inline packages/babel-traverse/README.md]: Nice. thanks.

**Comment by xtuc** [comment]
I started to write a documentation https://gist.github.com/xtuc/26931c424fb7148c000217de8e7fdf76 about how traversal and mutation works, could we try to integrate it? I think this should be a more general documentation.

**Comment by EmpireWorld** [comment]
@xtuc Sure, why not? We can provide a link to your full documentation, and also we can include some of your examples.

**Comment by xtuc** [comment]
My documentation can be included in the README. Simple examples for AST mutation are hard to find, I'm open if you have suggestions.

Let's merge this first :+1: ! We will need to backport it to the 7.0 branch.

File: packages/babel-traverse/README.md [hunk 2]
```diff
@@ -1 +1,30 @@
 # babel-traverse
+The Babel Traverse module maintains the overall tree state, and is responsible for replacing, removing, and adding nodes.
```
Comment (existentialism) [inline packages/babel-traverse/README.md]: Just for consistency and website parsing, let's add a newline and prefix with ">"

File: packages/babel-traverse/README.md [hunk 3]
```diff
@@ -1 +1,30 @@
...
+
+Install it by running:
```
Comment (existentialism) [inline packages/babel-traverse/README.md]: Can we add a heading to match other docs?

```markdown
## Install

``
$ npm install --save babel-traverse
``

```

File: packages/babel-traverse/README.md [hunk 4]
```diff
@@ -1 +1,30 @@
...
+```
+$ npm install --save babel-traverse
+```
+We can use it alongside Babylon to traverse and update nodes:
```
Comment (existentialism) [inline packages/babel-traverse/README.md]: Like above, add a heading:

```markdown
## Usage

// stuff

```

**Comment by hzoo** [comment]
Thanks

**Comment by EmpireWorld** [comment]
Please also update the npm README.md: https://www.npmjs.com/package/babel-traverse for new users.

**Pull request merged**